src-ghc7/Data/List/NonEmpty.hs

```diff
@@ -346,6 +350,24 @@ instance Foldable NonEmpty where
   toList = toList
 #endif
 
+#ifdef MIN_VERSION_template_haskell
+-- | @since 0.19
+instance TH.Lift a => TH.Lift (NonEmpty a) where
+    lift (x :| xs) = do
+        x' <- TH.lift x
+        xs' <- TH.lift xs
+        return $ TH.InfixE (Just x') (TH.ConE neConsName) (Just xs')
+      where
+        neConsName = TH.mkNameG_d semigroupsPackageKey "Data.List.NonEmpty" ":|"
+
+semigroupsPackageKey         :: String
+#ifdef CURRENT_PACKAGE_KEY
+semigroupsPackageKey          = CURRENT_PACKAGE_KEY
+#else
+semigroupsPackageKey          = "lens-0.19" -- TODO: either remember to update or use Paths_
```

Comment:
Yes, please do use `Paths_`—I don't know what this `lens` business is about :)

Reply:
Ah, package name have to be hardcoded anyway :S